app/assets/javascripts/application.js

```diff
@@ -104,12 +104,14 @@ if (!String.prototype.supplant) {
   // this is a bit hacky
   // TODO: To be moved to more page based controllers
```

Comment:
Discussed this offline with David and we decided that this PR should not address this right now, it would be a greater refactoring.